lock/client.go

```diff
@@ -14,7 +14,8 @@
 
 package lock
 
-type LockClient interface {
+// Client is a generic interface for a lock
+type Client interface {
```

Comment:
golint complains about "stuttering" which is the problem that I was trying to fix (callers outside the lock package refer to this as `lock.LockClient`)